third_party/move/move-model/src/builder/module_builder.rs

```diff
@@ -518,6 +519,9 @@ impl<'env, 'translator> ModuleBuilder<'env, 'translator> {
         let params = et.analyze_and_add_params(&def.signature.parameters, true);
         let result_type = et.translate_type(&def.signature.return_type);
         let kind = if def.entry.is_some() {
+            if et.env().language_version == LanguageVersion::V2_0 && def.inline {
```

Comment:
1. Why do we want to only check this if language version is 2.0? What is the behavior if language version is 1.0 (and is that acceptable behavior)? We should also document this.
2. Could we please create a helper function akin to `is_at_least(language_version)` that checks that the language version is at least some version, and not exactly some version? When we add a new language version, we will need to go all over the code base and update them, which is risky (as the person doing the language upgrade might not have all the context for similar gatekeeping all over the codebase).
3. Can we add a note somewhere in the Move book documenting this behavior? Ideally in the inline functions section or the entry section.

Reply:
The main reason for guarding this check is to make sure it is compatible with Language V1. For this specific this issue, V1 will crash in the extended check phase while V2 will compile before adding this check.

Reply:
Added an issue: https://github.com/aptos-labs/aptos-core/issues/13859 for your doc request, need to do it in a separate PR.

Reply:
@vineethk, I refactored the code according to your comment.

Reply:
Coming back to a part of question 1 more explicitly: if the language version is 1, and compiler version is 2, what is the expected behavior?

Reply:
It will pass compilation without an error or crash